module/modmanager/manager_test.go

```diff
@@ -0,0 +1,159 @@
+package modmanager
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/edaniels/golog"
+	"go.viam.com/test"
+	goutils "go.viam.com/utils"
+
+	"go.viam.com/rdk/components/generic"
+	"go.viam.com/rdk/components/motor"
+	"go.viam.com/rdk/config"
+	"go.viam.com/rdk/registry"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/testutils/inject"
+	"go.viam.com/rdk/utils"
+)
+
+func TestModManagerFunctions(t *testing.T) {
+	ctx := context.Background()
+	logger := golog.NewTestLogger(t)
+	modExe := utils.ResolveFile("examples/customresources/demos/simplemodule/run.sh")
+
+	myCounterModel := resource.NewModel("acme", "demo", "mycounter")
+	rNameCounter1 := resource.NameFromSubtype(generic.Subtype, "counter1")
+	cfgCounter1 := config.Component{
+		Name:  "counter1",
+		API:   generic.Subtype,
+		Model: myCounterModel,
+	}
+	_, err := cfgCounter1.Validate("test")
+	test.That(t, err, test.ShouldBeNil)
+
+	myRobot := &inject.Robot{}
+	myRobot.LoggerFunc = func() golog.Logger {
+		return logger
+	}
+
+	parentAddr, err := os.MkdirTemp("", "viam-test-*")
+	test.That(t, err, test.ShouldBeNil)
+	defer os.RemoveAll(parentAddr)
+	parentAddr += "/parent.sock"
+
+	myRobot.ModuleAddressFunc = func() (string, error) {
+		return parentAddr, nil
+	}
+
+	t.Run("AddModuleHelpers", func(t *testing.T) {
+		mgr, err := NewManager(myRobot)
+		test.That(t, err, test.ShouldBeNil)
+
+		mod := &module{name: "test", exe: modExe}
+
+		err = mod.startProcess(ctx, parentAddr, logger)
+		test.That(t, err, test.ShouldBeNil)
+
+		err = mod.dial()
+		test.That(t, err, test.ShouldBeNil)
+
+		err = mod.checkReady(ctx, parentAddr)
+		test.That(t, err, test.ShouldBeNil)
+
+		mod.registerResources(mgr, logger)
+		reg := registry.ComponentLookup(generic.Subtype, myCounterModel)
+		test.That(t, reg, test.ShouldNotBeNil)
+		test.That(t, reg.Constructor, test.ShouldNotBeNil)
+
+		err = mgr.Close(ctx)
+		test.That(t, err, test.ShouldBeNil)
+
+		registry.DeregisterComponent(generic.Subtype, myCounterModel)
+
+		reg = registry.ComponentLookup(generic.Subtype, myCounterModel)
+		test.That(t, reg, test.ShouldBeNil)
+	})
+
+	mgr, err := NewManager(myRobot)
+	test.That(t, err, test.ShouldBeNil)
+
+	t.Run("AddModule", func(t *testing.T) {
+		modCfg := config.Module{
+			Name:    "simple-module",
+			ExePath: modExe,
+		}
+		err := mgr.AddModule(ctx, modCfg)
```

Comment:
I'd be weary of making subtests that mutate one variable that gets used in the next. I'd suggest making a setup and teardown function that runs per sub test